sql/src/test/java/io/crate/analyze/CreateAlterTableStatementAnalyzerTest.java

```diff
@@ -1086,8 +1087,6 @@ public void testCreateTableGeneratedColumnWithMatch() {
 
     @Test
     public void testCreateTableGeneratedColumnBasedOnGeneratedColumn() {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage("A generated column cannot be based on a generated column");
```

Comment:
is it expected behaviour that this is now possible?

Reply:
I'd be in favour of reverting this behavior change and do it in a separate PR - together with change log entry + dedicated tests (also for the execution layer)

Reply:
nope, that was corrected, restoring original behaviour, apologies for the confusion

Reply:
Hi @mfussenegger,  just so that I have it correct:

```
 create table foo (
   ts timestamp with time zone,
   day as date_trunc('day', ts),
   date_string as cast(day as string))
```

you are saying that column `date_string` cannot be defined because is defined from day?
with my solution you may. Should I revert that?

Thank you

Reply:
I will revert this.